src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    }
+
+    sum += number;
+
+    return adder;
```

Comment:
The `adder` function works as expected, returning a function when a number is provided and the current sum when called without arguments. Make sure to test this thoroughly to ensure it handles all expected inputs correctly.